src/components/palette_item/palette_item.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
+import * as models from "../../models/shapes.ts";
+import {Circle} from "../shapes/Circle.tsx";
+import {Square} from "../shapes/Square.tsx";
+import {Triangle} from "../shapes/Triangle.tsx";
+
+export interface PaletteItemProps {
+  shapeModel: any;
```

Comment:
You can define constructor interfaces in TypeScript. IIRC, it's something like: `{new (x: number, y: number, size: number): models.Shape}`